airflow/models/dag.py

```diff
@@ -3967,18 +3980,33 @@ def _run_task(ti: TaskInstance, session):
     Args:
         ti: TaskInstance to run
     """
+    ret = None
     log.info("*****************************************************")
     if ti.map_index > 0:
         log.info("Running task %s index %d", ti.task_id, ti.map_index)
     else:
         log.info("Running task %s", ti.task_id)
     try:
-        ti._run_raw_task(session=session)
+        while True:
+            try:
+                ret = ti._run_raw_task(session=session, raise_on_defer=True)
+                break
+            except TaskDeferred as e:
```

Comment:
I wonder if we should add an argument to `_run_raw_task` so it skips all the leg work (mostly in `_defer_task` but also adding a Log). If we avoid `_defer_task` altogether, this block would not intefere with a real triggerer because the triggerer will never see the ti (being never in the DEFERRED state).